rotkehlchen/assets/asset.py

```diff
@@ -227,6 +227,10 @@ class AssetWithNameAndType(Asset, abc.ABC):
     asset_type: AssetType = field(init=False)
     name: str = field(init=False)
 
+    def get_asset_data(self) -> dict[str, Any]:
```

Comment:
1. why not make it a `TypedDict`?
2. why not move this method the `Asset` class?